src/service-api.ts

```diff
@@ -0,0 +1,106 @@
+// global
+import { FastifyPluginAsync } from 'fastify';
+
+// local
+import common, {
+  changePlan,
+  getPlans,
+  getOwnPlan,
+  getProrationPreview,
+  setDefaultCard,
+  getCurrentCustomer,
+} from './schemas';
+import { TaskManager } from './task-manager';
+import { Stripe } from 'stripe';
+import { Member } from 'graasp';
+import { CustomerExtra } from './interfaces/customer-extra';
+
+interface GraaspSubscriptionsOptions {
+  stripeSecretKey: string;
+  defaultPlanPriceId: string;
+}
+
+const plugin: FastifyPluginAsync<GraaspSubscriptionsOptions> = async (fastify, options) => {
+  const { stripeSecretKey, defaultPlanPriceId } = options;
+  const {
+    members: { dbService: mS, taskManager: memberTaskManager },
+    taskRunner: runner,
+  } = fastify;
+
+  const stripe = new Stripe(stripeSecretKey, { apiVersion: '2020-08-27' });
+
+  const taskManager = new TaskManager(mS, stripe);
+
+  runner.setTaskPreHookHandler<Member>(memberTaskManager.getCreateTaskName(), async (member) => {
+    const customer = await stripe.customers.create({ name: member.name, email: member.email });
+    const subscription = await stripe.subscriptions.create({
+      collection_method: 'charge_automatically',
+      customer: customer.id,
+      items: [{ price: defaultPlanPriceId }],
+    });
+    member.extra = { ...member.extra, customerId: customer.id, subscriptionId: subscription.id };
+  });
```

Comment:
Does this automatically charge the user? Or is the default free plan? Add comment.